src/DataProtection/custom/Cmdlets/Platform/Vault/Set-AzDataProtectionMSIPermission.ps1

```diff
@@ -0,0 +1,205 @@
+
+    process {
+          CheckResourcesModuleDependency
+          CheckPostgreSqlModuleDependency         
+          
+          $DatasourceId = $BackupInstance.Property.DataSourceInfo.ResourceId
+          $DatasourceType = $BackupInstance.Property.DataSourceInfo.ResourceType
+
+          if($DatasourceType -eq "Microsoft.Storage/storageAccounts"){$DatasourceType = "AzureBlob"}
+          elseif($DatasourceType -eq "Microsoft.Compute/disks"){$DatasourceType = "AzureDisk"}
+          elseif($DatasourceType -eq "Microsoft.DBforPostgreSQL/servers/databases"){$DatasourceType = "AzureDatabaseForPostgreSQL"}
+                 
+          $manifest = LoadManifest -DatasourceType $DatasourceType.ToString()
+
+          $ResourceArray = $DataSourceId.Split("/")
+          $ResourceRG = "/subscriptions/" + $ResourceArray[2] + "/resourceGroups/" + $ResourceArray[4]
+          $SubscriptionName = "/subscriptions/" + $ResourceArray[2]
+          $vault = Get-AzDataProtectionBackupVault -VaultName $VaultName -ResourceGroupName $VaultResourceGroup -SubscriptionId $ResourceArray[2]
+          
+          $AllRoles =  Get-AzRoleAssignment -ObjectId $vault.IdentityPrincipalId
+
+          $MissingRolesInitially = $false
+
+          foreach($Permission in $manifest.keyVaultPermissions)
+          {              
+              Write-Host "We will attempt to automatically grant the foloowing access: `n
+                            1.Backup vault’s identity access on the Postgres server and the key vault `n
+                            2.'Allow All Azure services' under network connectivity in the Key vault `n"
+
+              $Choice = Read-Host "Press n/N to exit the grant permissions or press any other key to continue"
+
+              if($Choice.ToLower() -eq "n")
+              {
+                  Write-Host "Cancelling Grant permissions operation"
+                  return
+              }
+
+              $ServerRG = $ResourceArray[4]
+              $ServerName = $ResourceArray[8] 
+
+              Update-AzPostgreSqlServer -ResourceGroupName $ServerRG -ServerName $ServerName -PublicNetworkAccess Enabled | Out-Null
+              New-AzPostgreSqlFirewallRule -Name AllowAllAzureIps -ResourceGroupName $ServerRG -ServerName $ServerName -EndIPAddress 0.0.0.0 -StartIPAddress 0.0.0.0 | Out-Null
+              
+              if($KeyvaultId -eq "")
+              {
+                  Write-Host "KeyvaultId not provided .Please note that permissions on keyvault may not be successfully assigned."  
+                  break
+              }
+                            
+              $KeyvaultArray = $KeyvaultId.Split("/")
+              $KeyvaultRG = "/subscriptions/" + $KeyvaultArray[2] + "/resourceGroups/" + $KeyvaultArray[4]
+              $KeyvaultSubscriptionName = "/subscriptions/" + $KeyvaultArray[2]
+              $KeyvaultName = $KeyvaultArray[8]
+              $KeyVault = Get-AzKeyVault -VaultName $KeyvaultName
+                                    
```

Comment:
Here we need to add a check ensuring that the KeyVault provided as the command argument indeed has the secret provided as the part of backup instance JSON object.